pilot/proxy/envoy/config.go

```diff
@@ -453,35 +452,10 @@ func buildTCPListener(tcpConfig *TCPRouteConfig, ip string, port int, protocol m
 				baseTCPProxy,
 			},
 		}
-	case model.ProtocolRedis:
-		// Redis filter requires the cluster name to be specified
-		// as part of the filter. We extract the cluster from the
-		// TCPRoute. Since TCPRoute has only one route, we take the
-		// cluster from the first route. The moment this route array
-		// has multiple routes, we need a fallback. For the moment,
-		// fallback to base TCP.
-
-		// Unlike Mongo, Redis is a standalone filter, that is not
-		// stacked on top of tcp_proxy
-		if len(tcpConfig.Routes) == 1 {
-			return &Listener{
-				Name:    fmt.Sprintf("redis_%s_%d", ip, port),
-				Address: fmt.Sprintf("tcp://%s:%d", ip, port),
-				Filters: []*NetworkFilter{{
-					Type: both,
-					Name: RedisProxyFilter,
-					Config: &RedisProxyFilterConfig{
-						ClusterName: tcpConfig.Routes[0].Cluster,
-						StatPrefix:  "redis",
-						ConnPool: &RedisConnPool{
-							OperationTimeoutMS: int64(RedisDefaultOpTimeout / time.Millisecond),
-						},
-					},
-				}},
-			}
-		}
```

Comment:
Instead of removing all the code, can you just change in the main switch statement to consider redis as TCP? And add comments in functions like above, stating that redis support has to be enabled in the said switch statement..

I don't want to add this code again, when we fix this.

Reply:
Note that the removed code in git anyway. In the future you can just revert this commit. Anyway, submitting a version without redis code removed.